email/src/main/java/edu/jhuapl/dorset/demos/EmailManager.java

```diff
@@ -0,0 +1,393 @@
+package edu.jhuapl.dorset.demos;
+
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.Properties;
+
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Part;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValue;
+
+public class EmailManager {
+    
+    private static final Logger logger = LoggerFactory.getLogger(EmailManager.class);
+    
+    private static final String USERNAME_KEY = "username";
+    private static final String PASSWORD_KEY = "password";
+    private static final String MAIL_STORE_TYPE_KEY = "mailStoreType";
+    private static final String HOST_KEY = "host";
+
+    private String user;
+    private String password;
+    private String mailStoreType;
+    private String host;
+    
+    private Session session;
+    private Store store;
+    private Folder inboxFolder;
+    private Folder errorFolder;
+    private Folder completeFolder;
+    private Message[] unreadMessages = new Message[1];
+    private Message[] processingMessages = new Message[1];
+    private ArrayDeque<Message> handler = new ArrayDeque<Message>();
+    private static final int HOLDS_MESSAGES = 1;
+    private String response = "";
+    
+    /**
+    * EmailManager Constructor
+    * 
+    * @param config  Configuration object that stores mail server information, username, and password
+    */
+    public EmailManager(Config config) {
+        
+        this.user = config.getString(USERNAME_KEY);
+        this.password = config.getString(PASSWORD_KEY);
+        this.mailStoreType = config.getString(MAIL_STORE_TYPE_KEY);
+        this.host = config.getString(HOST_KEY); 
+        Properties prop = extractProperties(config);
+        
+        try {
+            session = Session.getDefaultInstance(prop);
+            store = session.getStore(mailStoreType);
+            store.connect(host, this.user, this.password);
+        } catch (MessagingException e) {
+            logger.error("Failed to start email session");
+        }
+    }
+    
+    /**
+     * Extract properties from configuration object
+     * 
+     * @param config  Configuration object that stores mail server information, username, and password
+     * @return the properties file
+     */
+    private Properties extractProperties(Config config) {
+        Properties prop = new Properties();
+        for (java.util.Map.Entry<java.lang.String, ConfigValue> entry : config.entrySet()) {
+            prop.setProperty(entry.getKey(), config.getString(entry.getKey()));
+        }        
+        return prop;
+    }
+    /**
+     * Initialize 3 folders
```

Comment:
I wouldn't say "3 folders" here in a comment because in the future you may change this to 2 or 4 folders and then the comment goes out of date. Better to tell the developer what it does:
"Retrieves the folders and creates them if they don't exist"